src/platform/common/net/httpClient.ts

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+'use strict';
+
+import { inject, injectable } from 'inversify';
+import { parse, ParseError } from 'jsonc-parser';
+import { IHttpClient } from '../types';
+import { IServiceContainer } from '../../ioc/types';
+import { IWorkspaceService } from '../application/types';
+import { traceError, traceVerbose } from '../../logging';
+import * as fetch from 'cross-fetch';
+
+@injectable()
+export class HttpClient implements IHttpClient {
+    public readonly requestOptions: RequestInit;
+    constructor(@inject(IServiceContainer) serviceContainer: IServiceContainer) {
+        const workspaceService = serviceContainer.get<IWorkspaceService>(IWorkspaceService);
+        this.requestOptions = { headers: { proxy: workspaceService.getConfiguration('http').get('proxy', '') } };
+    }
+
+    public async downloadFile(uri: string): Promise<Response> {
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        return fetch.fetch(uri, this.requestOptions);
+    }
+
+    public async getJSON<T>(uri: string, strict: boolean = true): Promise<T> {
```

Comment:
Going to remove these. Not used anywhere.

Reply:
I removed the unit test for this class well because it's just a light wrapper around fetch. I don't think it needs a unit test